src/main/java/org/mule/extension/db/internal/resolver/param/StoredProcedureParamTypeResolver.java

```diff
@@ -57,31 +57,45 @@ public StoredProcedureParamTypeResolver(DbTypeManager dbTypeManager) {
   @Override
   public Map<Integer, DbType> getParameterTypes(DbConnection connection, QueryTemplate queryTemplate, List<ParameterType> types)
       throws SQLException {
+    ResultSet procedureColumns = null;
     DatabaseMetaData dbMetaData = connection.getJdbcConnection().getMetaData();
 
+    String schemaName = getStoreProcedureSchema(queryTemplate.getSqlText()).orElseGet(() -> "");
     String storedProcedureName = getStoredProcedureName(queryTemplate.getSqlText());
     if (dbMetaData.storesUpperCaseIdentifiers()) {
+      schemaName = schemaName.toUpperCase();
       storedProcedureName = storedProcedureName.toUpperCase();
     }
 
-    String storedProcedureSchema = getStoreProcedureSchema(queryTemplate.getSqlText()).orElse(null);
-    if (dbMetaData.storesUpperCaseIdentifiers() && storedProcedureSchema != null) {
-      storedProcedureSchema = storedProcedureSchema.toUpperCase();
-    }
+    try {
 
-    ResultSet procedureColumns =
-        dbMetaData.getProcedureColumns(connection.getJdbcConnection().getCatalog(), storedProcedureSchema, storedProcedureName,
-                                       "%");
+      procedureColumns =
+          dbMetaData.getProcedureColumns(schemaName.isEmpty() ? connection.getJdbcConnection().getCatalog() : schemaName,
+                                         connection.getJdbcConnection().getSchema(),
+                                         storedProcedureName, "%");
+      Map<Integer, DbType> paramTypes = getStoredProcedureParamTypes(connection, schemaName, procedureColumns);
+
+
+      //if still unable to resolve, remove all catalog and schema filters
+      //and use only sp name and column pattern.
+      if (paramTypes.isEmpty()) {
+        LOGGER.debug(String.format(
```

Comment:
You can use this `LOGGER.debug('"Failed to get procedure types with schema name {}  and procedure name {}. Removing all catalog and schema filters.",  schemaName, storedProcedureName)`